pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/ImportWrapper.java

```diff
@@ -93,7 +93,7 @@ public int hashCode() {
         if (name == null) {
             return fullname.hashCode();
         }
-        return name.hashCode();
+        return name.hashCode() + (isStaticDemand ? 1 : 0);
```

Comment:
When we change hashCode, we probably need to change equals as well.
With your change now, two instances of ImportWrapper could be equal (having the same name) but still give different hashCodes. That violates the general contract of hashCode/equals... https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/lang/Object.html#hashCode()

> If two objects are equal according to the equals(Object) method, then calling the hashCode method on each of the two objects must produce the same integer result. 

I guess, we first need to make sure, that wrapped "import" and "import static" are not equal. They may optionally have a different hash code (but that's just an optimization and not required). It is allowed, that two different (unequal) objects return the same hashCode (but it's not efficient - e.g. if put many of such objects into a hashmap, they would end up all in the same bucket and creating many collisions).

With your change of the hashCode function, you are relying on the implementation detail of "HashSet", that it actually uses this hashCode function.... which could change.